### PR TITLE
IMPR/GRID-265

### PR DESCRIPTION
**Pull request by Dwaynekj**
@joneit @rdepena 
Please reference the conversation at https://github.com/openfin/fin-hypergrid/pull/313
Paired PR with https://github.com/openfin/hyper-analytics/pull/29

### CODE REVIEW

**Comment by joneit** [comment]
@Dwaynekj could you please rebase this on top of develop?

**Comment by Dwaynekj** [comment]
Done

**Comment by Dwaynekj** [comment]
@joneit PR ready

File: src/dataModels/JSON.js [hunk 1]
```diff
@@ -363,45 +358,8 @@ var JSON = DataModel.extend('dataModels.JSON', {
...
-        return this.pipeline.find(function(pipe) {
-            var test = pipe.test,
-                type = typeof test;
-
-            test = type === 'function' && pipe.test ||
-                type === 'string' && this[pipe.test];
-
-            return test && test.call(this, event);
-        }.bind(this));
+        return this.dataSource.isDrillDown(event);
```
Comment (joneit) [inline src/dataModels/JSON.js]: replace `event` with `event.gridData.x` per my comment in https://github.com/openfin/hyper-analytics/pull/29#discussion-diff-78241725R275

Comment (Dwaynekj) [inline src/dataModels/JSON.js]: ok

File: src/Hypergrid.js [hunk 2]
```diff
@@ -3365,8 +3380,58 @@ Hypergrid.prototype = {
...
 
         return result;
-    },
+    }
 };
+function findOrCreateContainer(boundingRect) {
+    boundingRect = boundingRect || {};
+    boundingRect.width = boundingRect.width || '300px';
+    boundingRect.height = boundingRect.height || '300px';
+    boundingRect.position = boundingRect.position || 'relative';
```
Comment (joneit) [inline src/Hypergrid.js]: need `.top || 0` and `.left || 0`

File: src/Hypergrid.js [hunk 3]
```diff
@@ -3365,8 +3380,58 @@ Hypergrid.prototype = {
...
+    var div,
+        defaultID = 'hypergrid';
+    div = document.getElementById(defaultID);
+    if (div){
+        return div;
+    }
+    div = document.createElement('div');
+    div.setAttribute('id', defaultID);
+    div.style.width = boundingRect.width;
+    div.style.height = boundingRect.height;
```
Comment (joneit) [inline src/Hypergrid.js]: add need `div.style.left || 0 = boundingRect.left` and `div.style.top = boundingRect.top`

File: src/Hypergrid.js [hunk 4]
```diff
@@ -3365,8 +3380,58 @@ Hypergrid.prototype = {
...
+    div.setAttribute('ppsition', boundingRect.position);
```
Comment (joneit) [inline src/Hypergrid.js]: typo: p**p**ostiion

File: src/Hypergrid.js [hunk 5]
```diff
@@ -3365,8 +3380,58 @@ Hypergrid.prototype = {
         }
 
         return result;
-    },
+    }
 };
+function findOrCreateContainer(boundingRect) {
+    boundingRect = boundingRect || {};
+    boundingRect.width = boundingRect.width || '300px';
```
Comment (joneit) [inline src/Hypergrid.js]: let's make these default something slightly more practical like `500px` by `500px`

File: src/Hypergrid.js (hunk 2 above)
Comment (Dwaynekj) [inline src/Hypergrid.js]: done

File: src/Hypergrid.js (hunk 3 above)
Comment (Dwaynekj) [inline src/Hypergrid.js]: done

File: src/Hypergrid.js (hunk 4 above)
Comment (Dwaynekj) [inline src/Hypergrid.js]: fixed

File: src/Shared.js [hunk 6]
```diff
@@ -7,8 +7,8 @@
 module.exports = {
 
 	analytics: require('hyper-analytics'), // npm
-    //analytics: require('../../hyper-analytics'), // developers
-
+    // analytics: require('../../hyper-analytics'), // developers
```
Comment (joneit) [inline src/Shared.js]: git checkout src/Shared.js please

Comment (Dwaynekj) [inline src/Shared.js]: ok

**Pull request merged**